decryptor/mysql/prepared_statements.go

```diff
@@ -24,13 +24,15 @@ var ErrStatementNotFound = errors.New("no prepared statement with given statemen
 
 // PreparedStatementRegistry is a MySQL PreparedStatementRegistry.
 type PreparedStatementRegistry struct {
-	statements map[string]base.PreparedStatement
+	statements          map[string]base.PreparedStatement
+	querySelectSettings map[string][]*encryptor.QueryDataItem
```

Comment:
looks like querySelectSettings should be linked together with statement. lets force this constraint?
```
type registryEntry struct {
  statement base.PreparedStatement
  querySelectSettings []*encryptor.QueryDataItem
}
type PreparedStatementRegistry struct {
  items map[string]registryEntry  // or entries map[string]....
}
```
it will simplify removing pairs too